### PR TITLE
Adding module for Extensible Attributes object

### PR DESCRIPTION
**Pull request by matthewdennett**
This PR is to add the functionality of managing extensible attributes. 

- [x] Examples added to module 
- [x] Documentation added to module

This will address feature request:
https://github.com/infobloxopen/infoblox-ansible/issues/179

### CODE REVIEW

**Comment by matthewdennett** [comment]
Hey @hemanthKa677 can you please review this pull request?

**Comment by matthewdennett** [comment]
@ranjishmp @hemanthKa677 any chance this can get a review?

**Comment by ranjishmp** [comment]
@matthewdennett WE are getting ready with a major change for EA implementation to support searched based on EAs. Once that is in we will consider this PR

**Comment by matthewdennett** [comment]
@ranjishmp do you have a rough time frame on when that will be implemented?

**Comment by ranjishmp** [comment]
@matthewdennett @lkanthinfoblox is managing this now.

**Comment by hmalphettes** [comment]
@lkanthinfoblox I am also interested in this PR - thanks @matthewdennett for submitting it and for all the other contributions!

I had no problem rebasing this PR on the top of the latest codebase in my github. It is on my github fork.
I am able to configure or update extended attributes with this.

I would like to figure out how to configure the additional properties too eventually.

Let me know if there is anything I can do to help.

**Comment by matthewdennett** [comment]
@lkanthinfoblox, is it possible that this one be looked at please?

**Comment by lkanthinfoblox** [comment]
@hmalphettes and @matthewdennett, We started looking into this PR. We will update the outcome in couple of days!

**Comment by JkhatriInfobox** [comment]
Hello @matthewdennett,  I noticed that your pull request has been open for a while. Could you please update it with the latest changes from the `master` branch and resolve any conflicts if they exist? This will help us to review and merge your changes more efficiently.

**Comment by matthewdennett** [comment]
@JkhatriInfobox, fingers crossed I didn't break anything with that mege.

File: plugins/module_utils/api.py [hunk 1]
```diff
@@ -213,6 +214,12 @@ def convert_members_to_struct(member_spec):
         member_spec['members'] = [{'_struct': 'dhcpmember', 'name': k['name']} for k in member_spec['members']]
     return member_spec
```
Comment (hmalphettes) [inline plugins/module_utils/api.py:216]: From ansible sanity tests stable-2.16:

```
ERROR: Found 2 pep8 issue(s) which need to be resolved:
ERROR: plugins/module_utils/api.py:217:1: E302: expected 2 blank lines, found 1
ERROR: plugins/module_utils/api.py:224:1: E302: expected 2 blank lines, found 1
```

**Comment by hmalphettes** [comment]
Wow, thank you both.

The linting run by ansible sanity tests are reporting some missing new lines:
```
ERROR: Found 2 pep8 issue(s) which need to be resolved:
ERROR: plugins/module_utils/api.py:217:1: E302: expected 2 blank lines, found 1
ERROR: plugins/module_utils/api.py:224:1: E302: expected 2 blank lines, found 1
```

and also to use yield on a list in a different form for a file that is not touched by this pull request - only on [Sanity Tests (devel)](https://github.com/infobloxopen/infoblox-ansible/actions/runs/9277174884/job/25525816464?pr=186#logs)
https://github.com/infobloxopen/infoblox-ansible/blob/master/tests/unit/compat/mock.py#L67
https://github.com/infobloxopen/infoblox-ansible/blob/master/tests/unit/compat/mock.py#L94

```
ERROR: Found 2 pylint issue(s) which need to be resolved:
ERROR: tests/unit/compat/mock.py:67:8: use-yield-from: Use 'yield from' directly instead of yielding each element one by one
ERROR: tests/unit/compat/mock.py:94:12: use-yield-from: Use 'yield from' directly instead of yielding each element one by one
```

On my branch of Matthew's PR where I added support for configuring multiple values on the extensible attributes via the flags 
(https://github.com/matthewdennett/infoblox-ansible/compare/master...hmalphettes:infoblox-ansible:create-ext-attrs-mattdennett)
I added a commit to fix those linting issues.


I don't want to slow things down but if you want me to make a pull request or anything let me know!

**Comment by JkhatriInfobox** [comment]
Hi @matthewdennett 
We noticed a duplicate commit (fb83da309478d0fceacb50284d14b6b9cbefba0c) in your PR after last force push on your forked master. To maintain git history, please rebase your master and drop this commit, 
```
git rebase -i fb83da309478d0fceacb50284d14b6b9cbefba0c^
```
Replace “pick” with “drop” for the duplicate commit in the text editor that opens up.

**Please note:** This operation modifies the commit history, so please proceed with caution. If you’re unsure, consider creating a backup branch first or verify on local before pushing. Also, avoid force pushing after the rebase as it can disrupt the review process.

**Comment by hmalphettes** [comment]
@matthewdennett my thanks for putting in the support for flag.

The sanity tests point at some doc I wrote: `ERROR: plugins/modules/nios_extensible_attribute.py:85:24: unparsable-with-libyaml: None - mapping values are not allowed in this context`

Hopefully that explains the many more errors reported before that. I'll have a shot at fixing these later today.

**Comment by hmalphettes** [comment]
@matthewdennett these changes fixe the linting issues reported by the Sanity tests: https://github.com/hmalphettes/infoblox-ansible/commit/6af034df064fcd3f616a868d828ec469ea1093ea

**Comment by matthewdennett** [comment]
@hmalphettes awesome. I'll try and add those updates at some point tomorrow.

**Comment by hmalphettes** [comment]
@matthewdennett, we are left with one last line too long error.
Short story: the failing workflow is running against an older version of nios_next_ip.py
@JkhatriInfobox could you trigger a re-run? Otherwise we could always make yet another commit and push it to trigger that.

Here is the investigation if needed:

I checked out your code and ran the github actions from my github account: all is green.

I downloaded the artifacts attached to the latest failing github actions - https://github.com/infobloxopen/infoblox-ansible/actions/runs/9310510563/artifacts/1554330581 - and checked the failing `nios_next_ip.py` and it does not have the changes @matthewdennett  kindly added to fix the issue.

image

When I do the same on my successful run - https://github.com/hmalphettes/infoblox-ansible/actions/runs/9314551322 - we find the expected version of nios_next_ip.py

I am hoping it is just a cache issue and all we need is to trigger the github actions again.

**Comment by matthewdennett** [comment]
I noticed this too and made a few commits but nothing seemed to make it register the change.

**Comment by matthewdennett** [comment]
@JkhatriInfobox, can you please have a look at the issue we are seeing here?

**Comment by JkhatriInfobox** [comment]
Hi @matthewdennett, @hmalphettes,

We appreciate your PR. Unfortunately, it didn't pass some checks. To address this, we've initiated a new PR with necessary corrections and will be closing PR #186 and #228 once new PR reviewed and merged . Your efforts are truly valued and we invite you to check out the new PR #235.

Additionally, we've identified some functional issues which we plan to rectify in the new PR. We also noticed the absence of unit and integration tests, which we'll be adding.

Thanks again for your contribution.

**Comment by hmalphettes** [comment]
@JkhatriInfobox thanks for the update and the work of the team on this. Happy to contribute in any manner.

**Comment by JkhatriInfobox** [comment]
Closing this Pull Request as the same modifications have been successfully merged in PR https://github.com/infobloxopen/infoblox-ansible/pull/235. This action is taken to maintain the cleanliness of the project and prevent redundancy. Please feel free to reach out if you need any further assistance.

**Pull request closed without merging**